Documentation/guideline/csharp_coding_standards.md

```diff
@@ -112,24 +118,32 @@ The class definition contains class members in the following order, from *less*
 ### Cross-platform coding
 Our code should supports multiple operating systems. Don't assume we only run (and develop) on Windows. Code should be sensitvie to the differences between OS's. Here are some specifics to consider.
 
-* **DO** use `Enviroment.NewLine` instead of hard-coding the line break instead of `\r\n`, as Windows uses `\r\n` and OSX/Linux uses `\n`.
+* **DO** use `Enviroment.NewLine` instead of hard-coding the line break, as Windows uses `\r\n` and OSX/Linux uses `\n`.
 
-> **Note**
->
-> Be aware that thes line-endings may cause problems in code when using `@""` text blocks with line breaks.
+  > **Note**
+  >
+  > Be aware that these line-endings may cause problems in code when using `@""` text blocks with line breaks.
+  >
+  > ```cs
+  > var x = @"line1
+  > line2";
+  >
+  > Assert.AreEqual("line1\nline2", x); // may fail on Windows when checkout uses CRLF line endings
+  > Assert.AreEqual($"line1{Environment.NewLine}line2", x);
```

Comment:
`Environment.NewLine` does not solve the issue. `Environment.NewLine` is always `\r\n` on Windows, however the line endings of the file is configured by GIT config.
So if `@""` is used, the safest way is to assert: 
`Assert.AreEqual("line1\nline2", x.Replace("\r\n", "\n"));`
Or also use `@""`, which is at most cases keep the same line ending with the above `@""` in the same file:
```
Assert.AreEqual(@"
...", x);
```